fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricEntityTypeBuilder.java

```diff
@@ -269,6 +281,30 @@ protected FabricEntityTypeBuilder(SpawnGroup spawnGroup, EntityType.EntityFactor
 		return type;
 	}
 
+	public static class Minecart<T extends AbstractMinecartEntity> extends FabricEntityTypeBuilder<T> {
+		private MinecartComparatorLogic<T> customComparatorLogic = null;
+
+		protected Minecart(SpawnGroup spawnGroup, EntityType.EntityFactory<T> factory) {
```

Comment:
I personally am against this. With this, you need to override every superclass method that returns self to return a self with a correct type (i.e. type `Minecart` than `FabricEntityTypeBuilder`)

Reply:
So what do you propose instead? There are already two overrides in the same file that don't do this, so I'm not seeing how this is constructive unless you provide a solution.

Reply:
What's the point, can't people call `customComparatorLogic` before calling the other methods?

Reply:
I suggest something in the lines of https://github.com/FabricMC/fabric/blob/1.16/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricDefaultAttributeRegistry.java
i.e. people can register through entity type builder, yet they are allowed to associate comparator output to entity type freely

Reply:
I believe liach is right. It's simpler to directly expose `MinecartComparatorLogic` (we can drop the `Fabric` prefix imo).

Reply:
The reason I added the Living and mob subclass builders is that it puts all the code in one spot, vs registering and then forgetting to call a method later.

Reply:
The other facility exists fyi like with default attribute registries is to override those on existing entities.

Reply:
> and then forgetting to call a method later for entity attributes and spawn logic

That's a very weak point imo. Basing testing would show that it clearly doesn't work, and nothing stops you from forgetting to call the specialized builder method anyway.

Reply:
Especially with something as "unusual" as minecarts, maybe it makes more sense for entity attributes and spawn logic which are quite important to a lot of entities.

Reply:
I'm fine with splitting it from the entity type, but what module should I put it in if I do that?

Reply:
Still object builders. Notice how FabricDefaultAttributeRegistry sits right next to the entity type builder in object builders.